test/e2e/memcached_test.go

```diff
@@ -64,11 +64,55 @@ func TestMemcached(t *testing.T) {
 	}
 
 	t.Log("Creating new operator project")
+	var solveFailed bool
 	cmdOut, err := exec.Command("operator-sdk",
 		"new",
 		"memcached-operator").CombinedOutput()
 	if err != nil {
-		t.Fatalf("error: %v\nCommand Output: %s\n", err, string(cmdOut))
+		// HACK: dep cannot resolve non-master branches as the base branch for PR's,
+		// so running `dep ensure` will fail when first running
+		// `operator-sdk new ...`. For now we can ignore the first solve failure,
+		// and run `dep ensure` again.
+		// A permanent solution can be implemented once the following is merged:
+		// https://github.com/golang/dep/pull/1658
+		solveFailRe := regexp.MustCompile(`(?m)^[ \t]*Solving failure:.+github\.com/operator-framework/operator-sdk.+:$`)
+		if solveFailRe.Match(cmdOut) {
+			prSlug, ok := os.LookupEnv("TRAVIS_PULL_REQUEST_SLUG")
```

Comment:
I don't think it's necessary to duplicate this code. Since this has to be run at some point anyway, I think we should just run the gopkg fix immediately after the `new` command in all cases instead of later in the code. In that case, the regex check only contains a `t.Fatal` and we don't have duplicate code.